pkg/client/create_builder.go

```diff
@@ -348,15 +424,20 @@ func validateModule(kind string, module buildpack.BuildModule, source, expectedI
 	return nil
 }
 
-func uriFromLifecycleVersion(version semver.Version, os string, architecture string) string {
+func (c *Client) uriFromLifecycleVersion(version semver.Version, os string, architecture string) string {
 	arch := "x86-64"
 
 	if os == "windows" {
 		return fmt.Sprintf("https://github.com/buildpacks/lifecycle/releases/download/v%s/lifecycle-v%s+windows.%s.tgz", version.String(), version.String(), arch)
 	}
 
-	if architecture == "arm64" {
+	if builder.SupportedLinuxArchitecture(architecture) {
 		arch = architecture
+	} else {
+		// TODO: Do we want to fail in this case instead of a warning?
+		// I added the warning to help the buildpack authors understand if they builder ended up with a wrong
+		// lifecycle binary.
+		c.logger.Warnf("trying to download a lifecycle binary for an unsupported architecture: %s, using default %s", style.Symbol(architecture), style.Symbol(arch))
```

Comment:
I think a failure here would be appropriate

We might include some helpful text about how to find the right lifecycle binary

Reply:
The weird thing about failing is that **today** it doesn't fail it just returned the `x86-64` binary. Again, this is another behavior I didn't want to break, but sounds good to me at least to aware the end-user

Reply:
Could we make an issue for it?

Reply:
Captured in https://github.com/buildpacks/pack/issues/2163